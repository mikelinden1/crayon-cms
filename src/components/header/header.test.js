import Header from './header';
import sinon from 'sinon';

const props = {
    loggedIn: true,
    user: {
        name: 'Mike'
    },
    actions: {
        logout: () => {}
    }
}
it('Header matches snapshot', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
});

it('Header displays welcome correctly', () => {
    const wrapper = mountWrap(<Header {...props} />);
    const welcome = wrapper.find('.welcome').text();
    expect(welcome).toEqual(`Hi, ${props.user.name}!`);
});

it('Header calls logout', () => {
    const spy = sinon.spy();

    const propsWithSpy = {
        ...props,
        actions: {
            logout: spy
        }
    };

    const wrapper = mountWrap(<Header {...propsWithSpy} />);
    wrapper.find('.logout-btn').simulate('click');

    expect(spy.calledOnce).toBeTruthy();
});