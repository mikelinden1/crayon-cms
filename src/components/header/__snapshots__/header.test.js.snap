// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div>
  <header
    id="top-bar"
  >
    <Link
      replace={false}
      to="/"
    >
      <Icon
        icon={
          Object {
            "icon": Array [
              576,
              512,
              Array [],
              "f3fd",
              "M75.694 480a48.02 48.02 0 0 1-42.448-25.571C12.023 414.3 0 368.556 0 320 0 160.942 128.942 32 288 32s288 128.942 288 288c0 48.556-12.023 94.3-33.246 134.429A48.018 48.018 0 0 1 500.306 480H75.694zM512 288c-17.673 0-32 14.327-32 32 0 17.673 14.327 32 32 32s32-14.327 32-32c0-17.673-14.327-32-32-32zM288 128c17.673 0 32-14.327 32-32 0-17.673-14.327-32-32-32s-32 14.327-32 32c0 17.673 14.327 32 32 32zM64 288c-17.673 0-32 14.327-32 32 0 17.673 14.327 32 32 32s32-14.327 32-32c0-17.673-14.327-32-32-32zm65.608-158.392c-17.673 0-32 14.327-32 32 0 17.673 14.327 32 32 32s32-14.327 32-32c0-17.673-14.327-32-32-32zm316.784 0c-17.673 0-32 14.327-32 32 0 17.673 14.327 32 32 32s32-14.327 32-32c0-17.673-14.327-32-32-32zm-87.078 31.534c-12.627-4.04-26.133 2.92-30.173 15.544l-45.923 143.511C250.108 322.645 224 350.264 224 384c0 35.346 28.654 64 64 64 35.346 0 64-28.654 64-64 0-19.773-8.971-37.447-23.061-49.187l45.919-143.498c4.039-12.625-2.92-26.133-15.544-30.173z",
            ],
            "iconName": "tachometer-alt",
            "prefix": "fas",
          }
        }
      />
       Admin Dashboard
    </Link>
    <a
      href="http://localhost"
    >
      <Icon
        icon={
          Object {
            "icon": Array [
              576,
              512,
              Array [],
              "f015",
              "M488 312.7V456c0 13.3-10.7 24-24 24H348c-6.6 0-12-5.4-12-12V356c0-6.6-5.4-12-12-12h-72c-6.6 0-12 5.4-12 12v112c0 6.6-5.4 12-12 12H112c-13.3 0-24-10.7-24-24V312.7c0-3.6 1.6-7 4.4-9.3l188-154.8c4.4-3.6 10.8-3.6 15.3 0l188 154.8c2.7 2.3 4.3 5.7 4.3 9.3zm83.6-60.9L488 182.9V44.4c0-6.6-5.4-12-12-12h-56c-6.6 0-12 5.4-12 12V117l-89.5-73.7c-17.7-14.6-43.3-14.6-61 0L4.4 251.8c-5.1 4.2-5.8 11.8-1.6 16.9l25.5 31c4.2 5.1 11.8 5.8 16.9 1.6l235.2-193.7c4.4-3.6 10.8-3.6 15.3 0l235.2 193.7c5.1 4.2 12.7 3.5 16.9-1.6l25.5-31c4.2-5.2 3.4-12.7-1.7-16.9z",
            ],
            "iconName": "home",
            "prefix": "fas",
          }
        }
      />
       Site Home
    </a>
    <div
      className="grow"
    />
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        className="welcome"
      >
        Hi, 
        Mike
        !
      </div>
      <button
        className="logout-btn"
        onClick={[Function]}
      >
        <Icon
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f2f5",
                "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
              ],
              "iconName": "sign-out-alt",
              "prefix": "fas",
            }
          }
        />
         Logout
      </button>
    </div>
  </header>
  <div
    id="header-pad"
  />
</div>
`;
